### PR TITLE
Convert ShrineHome to use CustomScrollView

### PR DESCRIPTION
**Pull request by abarth**
This patch converts the Shrine home page to using a sliver-based grid.
This required using a CustomScrollView to mix the block at the top with
the grid below.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: examples/flutter_gallery/lib/demo/shrine/shrine_home.dart [hunk 1]
```diff
@@ -319,29 +374,27 @@ class _ShrineHomeState extends State<ShrineHome> {
...
+          ),
+          new SliverPadding(
+            padding: const EdgeInsets.all(16.0),
+            child: new SliverGrid(
+              gridDelegate: gridDelegate,
+              delegate: new SliverChildListDelegate(
+                _products.map((Product product) {
+                  return new ProductItem(
+                    product: product,
+                    onPressed: () { showOrderPage(product); }
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/shrine/shrine_home.dart]: all those commas everywhere and you missed the one on this line. :-)

**Comment by Hixie** [comment]
LGTM

File: examples/flutter_gallery/lib/demo/shrine/shrine_home.dart (hunk 1 above)
Comment (abarth) [inline examples/flutter_gallery/lib/demo/shrine/shrine_home.dart]: Fixed 🦅

**Pull request merged**